### PR TITLE
Various updates, module priorities, readdress and protect feature for infobot

### PR DESCRIPTION
**Pull request by ailin-nemui**
Hi there,

I have collected various commits, most of them should also be possible to apply individually. Please discuss if you are interested in taking them

- assorted bug fixes
- fixes when using the DBM::Deep store
- enable IPv6
- a feature to manually order Modules to ensure which ones take priority when catching messages
- a ways to protect facts in the infobot, so they can only be overwritten by authenticated users
- a ways to "ping" another user by having the infobot tell them about a fact, similar to "tell X about Y" but public instead of private message

### CODE REVIEW

File: lib/Bot/BasicBot/Pluggable/Module/Join.pm [hunk 1]
```diff
@@ -9,7 +9,7 @@ sub connected {
 
     ## If we are not a array reference, we are problably the old
     ## string format ... trying to convert
-    if ( not ref($channels) eq 'ARRAY' ) {
+    if ( not ref($channels) && $channels =~ 'ARRAY' ) {
```
Comment (bigpresh) [inline lib/Bot/BasicBot/Pluggable/Module/Join.pm]: I'm not quire sure I understand the point of this?

**Comment by bigpresh** [comment]
A rather excellent set of fixes and features, thank you!

I'll get them merged, possibly make a couple of tweaks, and get a new version out soon :)

File: lib/Bot/BasicBot/Pluggable/Module/Join.pm (hunk 1 above)
Comment (ailin-nemui) [inline lib/Bot/BasicBot/Pluggable/Module/Join.pm]: This fix is for DBM::Deep and similar backing stores; the channels are an object of DBM::Deep::Array which can be derived as array, but the "ref" function returns the class and not the storage type in that case

**Comment by bigpresh** [comment]
Right, 1.10 just went to CPAN, including these fixes and improvements - thank you very much again!

I'd like to apologise also, as I managed to release without including acknowledgement for them in the changelog - sorry about that!  I've added them now in the repo, so they'll be listed when the next release goes to CPAN at least.

**Pull request merged**